pyunlocbox/tests/test_functions.py

```diff
@@ -413,37 +410,59 @@ def test_independent_problems(self):
         # Test all available functions.
         funcs = inspect.getmembers(functions, inspect.isclass)
         for func in funcs:
+            print('Testing ' + func[0])
 
             # Instanciate the class.
             if func[0] in ['norm_tv']:
-                f = func[1](dim=1)  # Each column is one-dimensional.
+                f = func[1](
+                   dim=1, maxit=20, tol=0)  # Each column is one-dimensional.
             else:
                 f = func[1]()
 
+            # TODO make this test two dimensional for the norm nuclear?
+            exlude = ['func', 'norm_nuclear']
+
+            has_eval = ('_eval' in func[1].__dict__.keys() and
+                        not (func[0] in exlude))
+            has_prox = ('_prox' in func[1].__dict__.keys() and
+                        not (func[0] in exlude))
+            has_grad = ('_grad' in func[1].__dict__.keys() and
+                        not (func[0] in exlude))
+
             # The combined objective function of the N problems is the sum of
             # each objective.
-            if func[0] not in ['func', 'norm', 'norm_nuclear', 'proj']:
+            if has_eval:
                 res = 0
                 for iN in range(N):
                     res += f.eval(X[:, iN])
                 nptest.assert_array_almost_equal(res, f.eval(X))
 
             # Each column is the prox of one of the N problems.
             # TODO: norm_tv shoud pass this test. Why is there a difference ?
-            if func[0] not in ['func', 'norm', 'norm_nuclear', 'norm_tv',
-                               'proj']:
+            if has_prox:
```

Comment:
There's a method for this already. You can use `if 'PROX' in func[0].cap()`. Please do the same for `has_eval` and `has_grad` and remove those 3 variables.